pkgs/build-support/cc-wrapper/add-clang-cc-cflags-before.sh

```diff
@@ -6,6 +6,6 @@ for p in "${params[@]}"; do
     esac
 done
 
-if $needsTarget; then
+if $needsTarget && [[ $0 != *cpp ]]; then
```

Comment:
This is pretty opaque. Under what conditions would `$0` not match the result of shell expanding `*cpp` in the current directory? Why is that the correct test?

Reply:
I'm not sure, it's what worked to prevent weird errors with cpp on aarch64. Is there a way to make it match as expected but less opaque?

Reply:
> Under what conditions would `$0` not match the result of shell expanding `*cpp` in the current directory?

the shell is not expanding `*cpp` in the current directory. this is just a pattern match that `$0` doesn't end with cpp. `[[...]]` has different semantics than `[...]`

Reply:
The way it's opaque to me: 

1. What is `$0` in this context? Is this `add-clang-cc-cflags-before.sh`, run as a script, or something else?
2. Why is this contingent on matching exactly one thing with a `cpp` suffix? What's in the build directory such that this is an OK thing to do?
3. Why do these flags need to be skipped in this case?

As an implementation detail, I'm fine with this terse line. Without explanation, though, I find it completely inscrutable. I'd expect a huge comment in front of it explaining what's going on. 🙂

Reply:
> 1. What is `$0` in this context?

Afaik, `$0` is always going to be executable name in a shell script.

We're needing this patch because without it, libX11 was failing to build on aarch64 because `-march=armv8-a` was being included and the LLVM cpp does not like that so it errors out.



> 2\. Why is this contingent on matching exactly one thing with a `cpp` suffix?

Because it is n